taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
                             {
-                                value.map((tag, index) => {
+                                value.map((tagObj) => {
+                                    const index = tagObj['index']
+                                    const tag = tagObj['tag']
                                     return (
-                                        <Paper key={index} component={Stack} direction="column" sx={{ml: 4, width: 200}}>
-                                            <TextField 
-                                                size="small" 
-                                                sx={{mt: 1}} 
-                                                onChange = {event => {
-                                                    changeData(lang, index, event.target.value)
-                                                }}
-                                                value={tag} 
-                                                variant="outlined" />
-                                        </Paper>
+                                        <Stack key={index} direction="row" alignItems="center">
+                                            <Paper component={Stack} direction="column" sx={{ml: 4, width: 200}}>
+                                                <TextField 
+                                                    size="small" 
+                                                    sx={{mt: 1}} 
+                                                    onChange = {event => {
+                                                        changeData(lang, index, event.target.value)
+                                                    }}
+                                                    value={tag} 
+                                                    variant="outlined" />
+                                            </Paper>
+                                            <IconButton sx={{ml: 1, mt: 1, color: "#808080"}} onClick={() => handleDelete(lang, index)}>
+                                                 <DeleteOutlineIcon />
+                                            </IconButton>
+                                        </Stack>
                                     )
                                 })
                             }
                         </Box>
                     )
                 } )
             }
+            {/* Dialog box for adding translations */}
+            <Dialog open={open} onClose={handleClose}>
+                <DialogTitle>Add a language</DialogTitle>
+                <DialogContent>
+                <DialogContentText>
+                    Enter the two letter language code for the language to be added.
+                </DialogContentText>
+                <TextField
+                    autoFocus
+                    margin="dense"
+                    onKeyPress={(e) => { (e.key === 'Enter') && isValidLanguageCode && handleAddTranslation(newLanguageCode, e) }} 
```

Comment:
maybe better to use `charCode`, please check https://caniuse.com/?search=event.code